crowd_anki/config/config_settings.py

```diff
@@ -0,0 +1,74 @@
+from enum import Enum
+from dataclasses import dataclass
+
+from aqt import mw
+from ..utils.constants import USER_FILES_PATH
+
+@dataclass
+class ConfigSettings():
+    class Properties(Enum):
+        SNAPSHOT_PATH = "snapshot_path"
+        AUTOMATED_SNAPSHOT = "automated_snapshot"
+        SNAPSHOT_ROOT_DECKS = "snapshot_root_decks"
+        EXPORT_DECK_SORT_METHODS = "export_deck_sort_method"
+        EXPORT_DECK_SORT_REVERSED = "export_deck_sort_reversed"
+
+
+    class DeckExportSortMethods(Enum):
+        NO_SORTING = "none"
+        GUID = "guid"
+        FLAG = "flag"
+        TAG = "tag"
+        NOTEMODEL = "notemodel"
+        NOTEMODELID = "notemodelid"
+        FIELD1 = "field1"
+        FIELD2 = "field2"
+
+    snapshot_path: str
+    automated_snapshot: bool
+    snapshot_root_decks: list
+
+    export_deck_sort_reversed: bool
+    export_deck_sort_methods: list
+
+
+    def __init__(self, get_settings = True):
+        if get_settings:
+            self.config_file = mw.addonManager.getConfig(__name__)
```

Comment:
I believe this is problematic. This means that we cache the settings on the creation of the object (which would happen on plugin startup, at least in some cases), so to get new values a person would have to restart Anki, which is not a great experience. 

I think the config_file should be a cached property with a small ttl (https://pypi.org/project/cached-property/) or even just a function that gets the file each time

Reply:
The ConfigSettings class is the only way to read the config values, yes. But it is also the only way to set them yourself now. Opening the config options directly interacts with this class, and everywhere in the code relies on this class to give the values. It is made in the main and passed everywhere else, so it is pretty much a global singleton. If I have missed something in this implementation then I can go back and rethink this, but I do not currently see a time where a users value could be cached so that their intended value as not set. Unless they change the settings json with some other file and not through Anki? But 1. Do we wish to support that? and 2. the settings in the config window will not reflect their changes so it will be easy to see nothing has changed.

Reply:
The thing is - no one prevents people from creating multiple instances of the class. 's not actually a singletone (though making it such can be a solution).

A simple example of how it breaks: 
* Launch anki, go to config, enable automated snapshot. 
* What I expect to happen: when I quit anki/switch profile automated snapshot happens
* What actually happens: automated snapshot does not happen until the Anki is restarted and the new config values are loaded.

The previous config usages that I had were broken in a similar fashion on reflection. 

Some solutions for this would be to:
* do the cached property thing I mentioned above 
* make the config class an actual singleton

Reply:
what is you current thinking on this?